Fixes #35279 - Add SLES support for host registration

sbernhard opened the pull request:
<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

sbernhard committed:
Fixes #35279 - Add SLES support for host registration

stejskalleos commented on app/views/unattended/provisioning_templates/registration/global_registration.erb:
```diff
@@ -10,6 +10,9 @@ description: |
 #!/bin/sh
+
+# Make sure, all command output can be parsed (e.g. from subscription-manager)
+export LANG=en_US
```
can we also export `LC_ALL=C LANG=C`, see https://github.com/theforeman/foreman/pull/9261#discussion_r920123002

sbernhard replied:
done. Thanks.

stejskalleos commented on app/views/unattended/provisioning_templates/registration/global_registration.erb:
```diff
@@ -57,18 +60,27 @@ echo '#'
+  cat << EOF > /tmp/foreman_registration.repo
 [foreman_register]
 name=foreman_register
 baseurl=<%= shell_escape @repo %>
```
In the UI Form [0] and API [1] are examples for the Red Hat Family and Debian family.
Can we add examples also for the Suse/SLES family?

Note: Would be also nice to update the docs [2], we don't have example there at all. 

[0] https://github.com/theforeman/foreman/blob/develop/webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/Repository.js
[1] https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/registration_commands_controller.rb#L22
[2] https://docs.theforeman.org/nightly/Managing_Hosts/index-foreman-el.html#Customizing_the_Registration_Templates_managing-hosts

sbernhard replied:
Done. I will ask my colleage to have a look at the docs.

stejskalleos commented on app/controllers/api/v2/registration_commands_controller.rb:
```diff
@@ -13,7 +13,7 @@ class RegistrationCommandsController < V2::BaseController
         param :operatingsystem_id, :number, desc: N_("ID of the Operating System to register the host in. Operating system must have a `host_init_config` template assigned")
         param :smart_proxy_id, :number, desc: N_("ID of the Smart Proxy. This Proxy must have enabled both the 'Templates' and 'Registration' features")
-        param :setup_insights, :bool, desc: N_("Set 'host_registration_insights' parameter for the host. If it is set to true, insights client will be installed and registered on Red Hat family operating systems")
+        param :setup_insights, :bool, desc: N_("Set 'host_registration_insights' parameter for the host. If it is set to true, insights client will be installed and registered on Red Hat and SUSE family operating systems")
```
Shouldn't be this change for the `:repo` param instead of `:setup_insights`?

stejskalleos commented on webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/Repository.js:
```diff
@@ -21,7 +21,7 @@ const Repository = ({
         <LabelIcon
           text={__(
-            "A repository to be added before the registration is performed. It can be useful to e.g. make the subscription-manager packages available for the purpose of the registration. For Red Hat family distributions, this should be the URL of the repository, e.g. 'http://rpm.example.com/'. For Debian OS families, it's the whole list file content, e.g. 'deb http://deb.example.com/ buster 1.0'. ...
+            "A repository to be added before the registration is performed. It can be useful to e.g. make the subscription-manager packages available for the purpose of the registration. For Red Hat and SUSE family distributions, this should be the URL of the repository, e.g. 'http://rpm.example.com/'. For Debian OS families, it's the whole list file content, e.g. 'deb http://deb.example.com/ bus ...
```
JS tests are failing because of outdated snapshots, you can update the by running:
```
npm test -- -u --collectCoverage=false --silent --testPathPattern=./webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/
```

stejskalleos reviewed (changes requested):
Few thing plus it looks like GPG check for zypper repo is not enabled.

Registration command
```
curl -sS --insecure --user admin:changeme 'http://centos8-katello-devel-stable.example.com/register?activation_keys=key1&location_id=2&operatingsystem_id=2&organization_id=1&repo=http%3A%2F%2Fdownload.virtualbox.org%2Fvirtualbox%2Frpm%2Fel%2F7%2Fx86_64&repo_gpg_key_url=https%3A%2F%2Fwww.virtualbox.org%2Fdownload%2Foracle_vbox.asc&update_packages=false'  | bash
```

In SLES (mine is 12.3):
```bash
zypper repos
=>
# | Alias                 | Name                                           | Enabled | GPG Check | Refresh
--+-----------------------+------------------------------------------------+---------+-----------+--------
3 | foreman_register      | foreman_register                               | Yes     | (  ) No   | No     

cat /etc/zypp/repos.d/foreman_register.repo

[foreman_register]
name=foreman_register
enabled=1
autorefresh=0
baseurl=http://download.virtualbox.org/virtualbox/rpm/el/7/x86_64
type=NONE
gpgcheck=
```

On Stream 8 it works fine:
```
cat /etc/yum.repos.d/foreman_registration.repo
[foreman_register]
name=foreman_register
baseurl=http://download.virtualbox.org/virtualbox/rpm/el/7/x86_64
enabled=1
gpgcheck=1
gpgkey=https://www.virtualbox.org/download/oracle_vbox.asc
```

sbernhard commented:
I had a look at a SLES 12.3 again and adding the repo with "zypper addrepo" worked. Can you please share the /tmp/foreman_registration.repo and paste the zypper version?

Additionally, I added "type=rpm-md" - I hope this isn't a problem on RHEL OS familiy. 

Thanks for the other findings. I fixed them.

stejskalleos reviewed (approved):
:green_apple: **LGTM**
There is an issue [0] with adding repo on debian but it's not caused changes from this PR, let's get this in!

[0] https://projects.theforeman.org/issues/35342

stejskalleos commented:
Thanks @sbernhard !
